### PR TITLE
Enable UT for metadata controller and Open entity type

### PR DESCRIPTION
**Pull request by xuzhg**
### Issues

* Enable UT for metadata controller and Open entity type

### Description

* Enable UT for metadata controller and Open entity type

### Checklist (Uncheck if it is not completed)

- [ ] *Test cases added*
- [ ] *Build and test with one-click build and test script passed*

### Additional work necessary

*If documentation update is needed, please add "Docs Needed" label to the issue and provide details about the required document change in the issue.*

### CODE REVIEW

File: test/UnitTest/Microsoft.Test.AspNet.OData/MetadataControllerTest.cs [hunk 1]
```diff
@@ -936,26 +855,26 @@ public void GetMetadata_Returns_EdmModelFromRequest()
...
+            var response = await client.GetAsync("http://localhost/odata/$metadata");
             var responseString = await response.Content.ReadAsStringAsync();
 
             // Assert
             Assert.Contains("CustomizeNamepace.GetNS", responseString);
             Assert.Contains("Namespace=\"CustomizeNamepace\"", responseString);
         }
 
+#if NETFX
         [Fact]
```
Comment (robward-ms) [inline test/UnitTest/Microsoft.Test.AspNet.OData/MetadataControllerTest.cs]: Ditto here.

File: test/UnitTest/Microsoft.Test.AspNet.OData/OpenEntityTypeTest.cs [hunk 2]
```diff
@@ -123,14 +149,17 @@ public class OpenEntityTypeTest
...
+            {
+                config.Count();
+                config.MapODataServiceRoute("odata", "odata", GetUntypedEdmModel());
+            });
+            var client = TestServerFactory.CreateClient(server);
 
             // Act
             HttpResponseMessage response = await client.GetAsync(_untypedCustomerRequestRooturl + requestUri);
-
+            string a = response.Content.ReadAsStringAsync().Result;
```
Comment (robward-ms) [inline test/UnitTest/Microsoft.Test.AspNet.OData/OpenEntityTypeTest.cs]: is this needed?

Comment (xuzhg) [reply test/UnitTest/Microsoft.Test.AspNet.OData/OpenEntityTypeTest.cs]: Not needed. Thanks

File: test/UnitTest/Microsoft.Test.AspNet.OData/MetadataControllerTest.cs [hunk 3]
```diff
@@ -29,18 +44,18 @@ public class MetadataControllerTest
...
-            var response = await client.GetAsync("http://localhost/$metadata");
+            var response = await client.GetAsync("http://localhost/odata/$metadata");
 
             // Assert
             Assert.True(response.IsSuccessStatusCode);
             Assert.Equal("application/xml", response.Content.Headers.ContentType.MediaType);
             Assert.Contains("<edmx:Edmx", await response.Content.ReadAsStringAsync());
         }
 
+#if NETFX
```
[the remark this answers is not in the document]
Comment (xuzhg) [reply test/UnitTest/Microsoft.Test.AspNet.OData/MetadataControllerTest.cs]: A comment added. Thanks.

File: test/UnitTest/Microsoft.Test.AspNet.OData/MetadataControllerTest.cs (hunk 1 above)
Comment (xuzhg) [reply test/UnitTest/Microsoft.Test.AspNet.OData/MetadataControllerTest.cs]: Move this test cases above.

**Pull request merged**